File: programmers/level2/더_맵게.java
package level2;

import java.util.*;

public class 더_맵게 {
	public static int solution(int[] scoville, int K) {
		int answer = 0;
		PriorityQueue<Integer> queue = new PriorityQueue<Integer>();
		for (int i : scoville) {
			queue.add(i);
		}

		while (queue.peek() < K) {
			if(queue.size()==1) {
				return -1;
			}
			int n = queue.poll();
			n += queue.poll() * 2;
			queue.add(n);
			answer++;
		}
		return answer;
	}

	public static void main(String[] args) {
		System.out.println(solution(new int[] { 1, 2, 3, 9, 10, 12 }, 7));
	}
}

File: programmers/level2/더 맵게.py
import heapq


def solution(scoville, K):
    answer = 0
    heapq.heapify(scoville)

    while len(scoville) > 1:
        if scoville[0] >= K:
            return answer
        s = heapq.heappop(scoville) + heapq.heappop(scoville) * 2
        heapq.heappush(scoville, s)
        answer += 1
    return answer if scoville[0] >= K else -1


print(solution([1, 2, 3, 9, 10, 12], 7))
print(solution([1, 2, 3, 9, 10, 12], 9999999))
